book/src/pong_tutorial/pong_tutorial_01.md

```diff
@@ -0,0 +1,171 @@
+use amethyst::renderer::{DisplayConfig, DrawFlat, Event, KeyboardInput,
+                         Pipeline, PosTex, RenderBundle, RenderSystem,
+                         Stage, VirtualKeyCode, WindowEvent};
+```
+
+We'll be learning more about these as we go through this tutorial. The prelude 
+includes the basic (and most important) types like `Application`, `World`, and 
+`State`.
+
+Now we create our core game struct:
+
+```rust,ignore
+struct Pong; 
+```
+
+We'll be implementing the [`State`][st] trait on this struct, which is used by 
+Amethyst's state machine to start, stop, and update the game. But for now we'll 
+just implement one method: 
+
+```rust,ignore
+impl State for Pong {
+    fn handle_event(&mut self, _: &mut World, event: Event) -> Trans {
+        match event {
+            Event::WindowEvent { event, .. } => match event {
+                WindowEvent::KeyboardInput {
+                    input:
+                        KeyboardInput {
+                            virtual_keycode: Some(VirtualKeyCode::Escape),
+                            ..
+                        },
+                    ..
+                } => Trans::Quit,
+                _ => Trans::None,
+            },
+            _ => Trans::None,
+        }
+    }
+}
+```
+
+The `handle_event` method is executed on every frame before updating, and it's 
+used to react to events. It returns a `Trans`, which is an enum of state machine 
+transitions. In this case, we're watching for the Escape keycode from the 
+Window. If we receive it, we'll return `Trans::Quit` which will be used to clean 
+up the `State` and close the application. All other keyboard input is ignored 
+for now.
+
+Now that we know we can quit, let's add some code to actually get things 
+started! We'll use a `run()` function, which allows for bettor Error handling:
+
+```rust,ignore
+fn run() -> Result<()> {
+
+  //We'll put the rest of the code here
+
+  Ok(())
+}
+```
+
+Inside run() we first define a path for our display_config.ron file and load it.
+
+```rust,ignore
+let path = format!(
+    "{}/resources/display_config.ron",
+    env!("CARGO_MANIFEST_DIR")
+);
+let config = DisplayConfig::load(&path);
+```
+
+This .ron file was automatically generated by `amethyst new`. If you didn't use 
+`amethyst new`, now would be a good time to create this config file inside a 
+folder named resources. If you already have this file, we have some changes to 
+make, anyway:
+
+```ron,ignore
+(
+  title: "Pong!",
+  dimensions: Some((500, 500)),
+  max_dimensions: None,
+  min_dimensions: None,
+  fullscreen: false,
+  multisampling: 1,
+  visibility: true,
+  vsync: true,
+)
+```
+
+This will set the default window dimensions to 500 x 500, and make the title bar 
+say "Pong!" instead of the sad, lowercase default of "pong".
+
+Now, back inside our run() function in main.rs, let's copy and paste some 
+rendering code we don't fully understand, just so we can keep moving. The next 
```

Comment:
Maybe replace this with `(you don't need to understand what's going on in this rendering code yet.)`

The current version makes it sound like no one on Amethyst knows why this works :P

Reply:
I realize now that my promise of "The next chapter will cover rendering more in depth" as pertains to this chunk of code wasn't exactly accurate. Should we try to add an explanation of the rendering pipeline to chapter 1 or chapter 2? Or maybe introduce a brief chapter 3, or point to some other reference doc, to explain the "magic" of this setup before we move on?